Tests/CommandsTests/PackageCommandTests.swift

```diff
@@ -48,10 +48,9 @@ final class PackageCommandTests: CommandsTestCase {
         XCTAssertMatch(stdout, .contains("USAGE: swift package"))
     }
 
-    func testUsage() async throws {
-        throw XCTSkip("rdar://131126477")
```

Comment:
Do we have this radar resolved then? I find it confusing that any option that starts with `-h` leads to this unexpected behavior. We should at least raise this on the SAP repo for them to resolve this first.

Reply:
I added a comment on the internal radar saying basically I think this is works as intended, but maybe @natecook1000 can chime in.

Reply:
cc @rauhul

Reply:
I'm surprised that's the default, given that both `swiftc` and `clang` stick to single dash options, and thus `clang -fblocks` is clearly not equivalent to `clang -f -b -l -o -c -k -s`.

Reply:
swiftc and clang also use options with `-long-name` so I wouldn't take them as amazing prior art. arg parser follows the pattern of `ls -asl` and `git clean -Xdf` where short names are combined.

Reply:
My experience with this type of combined syntax is as a POSIX convention. See Guideline #5 in the [POSIX Utility Conventions](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap12.html).

Reply:
@MaxDesiatov @rauhul Any lingering concerns with this PR/behaviour?

Reply:
I'm happy with it

Reply:
Even if `swift build -halp` is parsed the same as `swift build -h -a -l -p` and `-h` is a valid flag, I'd still expect an error message to be printed about invalid `-a -l -p` flags.

Reply:
I agree with @MaxDesiatov .

I also think swift-argument-parser test need to be augmented to cover this case, if they currently don't!